docs/specification/draft/basic/authorization.mdx

```diff
@@ -150,6 +177,23 @@ these authorization servers, MCP clients will have to either:
    OAuth client themselves (e.g., through a configuration interface hosted by the
    server).
 
+#### Scope Selection Strategy
+
+When implementing authorization flows, MCP clients **SHOULD** follow the principle of least privilege by requesting
+only the scopes necessary for their intended operations. During the initial authorization handshake, MCP clients
+**SHOULD** follow this priority order for scope selection:
+
+1. **Use `scopes_default`** from the Protected Resource Metadata document as the starting point
+2. **If `scopes_default` is not available**, use all scopes defined in `scopes_supported`
```

Comment:
Wouldn't requesting/issuing a token with all the supported scopes result in an overly scoped token that increases the security risk, in case the token is compromised?

Reply:
@ayshsandu - if the server doesn't provide any other clues as to what scopes should be requested, the client alone can't determine the "minimum required" and has to use what's supported.